app_feup/lib/main.dart

```diff
@@ -90,7 +89,23 @@ class MyAppState extends State<MyApp> {
   @override
   void initState() {
     super.initState();
-    Timer.periodic( Duration(seconds: 60),
-        (Timer t) => state.dispatch( SetCurrentTimeAction(DateTime.now())));
+    Timer.periodic(Duration(seconds: 60),
+        (Timer t) => state.dispatch(SetCurrentTimeAction(DateTime.now())));
+  }
+
+  Route makePageTransition(
+      {Widget page, bool maintainState = true, RouteSettings settings}) {
+    return PageRouteBuilder(
+        pageBuilder: (BuildContext context, Animation<double> animation,
+            Animation<double> secondaryAnimation) {
+          return page;
+        },
+        transitionDuration: Duration(milliseconds: 200),
+        settings: settings,
+        maintainState: maintainState,
+        transitionsBuilder: (BuildContext context, Animation<double> animation,
+            Animation<double> secondaryAnimation, Widget child) {
+          return FadeTransition(opacity: animation, child: child);
+        });
```

Comment:
I think this builder should be extracted

Reply:
Hey! Where exactly would you put this funcition?